### PR TITLE
feat(cli): Add functionality to operate on Variables

### LINKED ISSUE #301: CLI: Add functionality to operate on Variables
## Description

We would like to perform the following operations for variable using our CLI:

- List all variables under a project (`keyshade variable list <projectId>`)
- Fetch all revisions of a variable (`keyshade variable revisions <variableId>`)
- Create a variable (`keyshade variable create <projectId>(refer postman for parameters)`)
- Update a variable (`keyshade variable update <variableId> (refer postman for parameters)`)
- Rollback a variable (`keyshade variable rollback <variableId> --environment-slug/-e <> --version/-v <>`)
- Delete a variable (`keyshade variable delete <variableId>`)

## Solution

- Create an implementation of `command.interface.ts` and name if `VariableCommand`.
- Stash all the functions in `src/commands/variable`.
- Use the `variableController` from `ControllerInstance` to make the API calls.

### PR DESCRIPTION
**Pull request by anudeeps352**
## Description

Add functionality to operate on Variables <br />

Fixes #301 

## Developer's checklist

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-check on my work

**If changes are made in the code:**

- [X] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [X] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [X] There are no UI/UX issues

### Documentation Update

- [X] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
Enhancement

___

### **Description**
- Added a new `VariableCommand` class to manage variables through the CLI, including subcommands for creating, deleting, listing, fetching revisions, rolling back, and updating variables.
- Implemented individual command classes for each variable operation, defining necessary arguments and options.
- Integrated the `VariableCommand` into the CLI application, making it available for use.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>variable.command.ts</strong><dd><code>Add `VariableCommand` class for CLI variable operations</code>&nbsp; &nbsp; </dd></summary>
<hr>

apps/cli/src/commands/variable.command.ts

<li>Introduced <code>VariableCommand</code> class extending <code>BaseCommand</code>.<br> <li> Added subcommands for variable management.<br> <li> Provides descriptions and names for the command.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/514/files#diff-ca846d5455581c0685b05ff3ba897751f59cd69d930f2e2a962e8fe1dfceae5d">+28/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>create.variable.ts</strong><dd><code>Implement `CreateVariable` command for CLI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/cli/src/commands/variable/create.variable.ts

<li>Implemented <code>CreateVariable</code> class for creating variables.<br> <li> Defined command arguments and options.<br> <li> Added action method to handle variable creation.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/514/files#diff-bc3afb04230c3c629f17dbd1fc5f734e65134bf61ad18ea26d65998b948b9ff7">+119/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>delete.variable.ts</strong><dd><code>Implement `DeleteVariable` command for CLI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/cli/src/commands/variable/delete.variable.ts

<li>Implemented <code>DeleteVariable</code> class for deleting variables.<br> <li> Defined command arguments.<br> <li> Added action method to handle variable deletion.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/514/files#diff-e16796e767a5307bc3bfa741cf14d20361c970e53055d4051faca8211b66a49f">+44/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>list.variable.ts</strong><dd><code>Implement `ListVariable` command for CLI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/cli/src/commands/variable/list.variable.ts

<li>Implemented <code>ListVariable</code> class for listing variables.<br> <li> Defined command arguments.<br> <li> Added action method to fetch and display variables.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/514/files#diff-10b95c97a99af739c47041522cb44cdc5bb784e26faa42731ea78b27e894c9c8">+50/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>revisions.variable.ts</strong><dd><code>Implement `FetchVariableRevisions` command for CLI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/cli/src/commands/variable/revisions.variable.ts

<li>Implemented <code>FetchVariableRevisions</code> class for fetching variable <br>revisions.<br> <li> Defined command arguments.<br> <li> Added action method to retrieve and display revisions.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/514/files#diff-0418e51a07f5eb276c22c750a6ff91907cc2578e7225282cc8d7df32b8a64a18">+63/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rollback.variable.ts</strong><dd><code>Implement `RollbackVariable` command for CLI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/cli/src/commands/variable/rollback.variable.ts

<li>Implemented <code>RollbackVariable</code> class for rolling back variables.<br> <li> Defined command arguments and options.<br> <li> Added action method to perform rollback operation.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/514/files#diff-fe376d59915c6c3b8ee0fc07aa4b1e6ef306f24d21439e4be6f1bf705a071589">+78/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>update.variable.ts</strong><dd><code>Implement `UpdateVariable` command for CLI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/cli/src/commands/variable/update.variable.ts

<li>Implemented <code>UpdateVariable</code> class for updating variables.<br> <li> Defined command arguments and options.<br> <li> Added action method to handle variable updates.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/514/files#diff-e081dceac29d37193bb60782e1cbfffc930128f4f0d752af0946baac3256e776">+69/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Integrate `VariableCommand` into CLI application</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/cli/src/index.ts

<li>Imported <code>VariableCommand</code> into the CLI application.<br> <li> Added <code>VariableCommand</code> to the list of available commands.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/514/files#diff-4565ec586fe8103913bc7f7abc9b548cb8f4d8403669752a5f10874118093bce">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: apps/cli/src/commands/variable/list.variable.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      )
+
+    if (success) {
+      const variable = data
+      if (variable.length > 0) {
+        data.forEach((variable: any) => {
+          Logger.info(`- ${variable.name} (${variable.value})`)
+        })
+      } else {
+        Logger.info('No variable found')
```
Comment (rajdip-b) [inline apps/cli/src/commands/variable/list.variable.ts:44]: variables**

**Review by rajdip-b — CHANGES_REQUESTED** [review]

**Review by rajdip-b — APPROVED** [review]

File: apps/cli/src/commands/variable/create.variable.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      const entryObj: { value: string; environmentSlug: string } = {
+        value: '',
+        environmentSlug: ''
+      }
+      entry.split(' ').forEach((pair) => {
+        const [key, value] = pair.split('=')
+        entryObj[key] = value
+      })
+      return entryObj
+    })
```
[the remark this answers is not in the document]
Comment (rajdip-b) [reply apps/cli/src/commands/variable/create.variable.ts:111]: @anudeeps352 you missed this one

[the remark this answers is not in the document]
Comment (anudeeps352) [reply apps/cli/src/commands/variable/create.variable.ts:111]: Ahh man lol. I'm down with fever thats why lol

[the remark this answers is not in the document]
Comment (rajdip-b) [reply apps/cli/src/commands/variable/create.variable.ts:111]: Take rest bro no rush!

[the remark this answers is not in the document]
Comment (anudeeps352) [reply apps/cli/src/commands/variable/create.variable.ts:111]: So just make changes according to the recommendation right ?

**Review by rajdip-b — APPROVED** [review]

**Pull request merged**